apps/checkout/src/sections/Contact/GuestUserForm.tsx

```diff
@@ -54,8 +59,21 @@ export const GuestUserForm: React.FC<AnonymousCustomerFormProps> = ({
 
   const [, updateEmail] = useCheckoutEmailUpdateMutation();
 
-  const onSubmit = ({ email }: FormData) =>
-    updateEmail({ id: checkout.id, email });
+  const onSubmit = async ({ email }: FormData) => {
+    const result = await updateEmail({
+      email,
+      checkoutId: checkout.id,
+    });
+
+    const [hasErrors, errors] = extractMutationErrors(result);
+
+    if (!hasErrors) {
+      showSuccess();
+      return;
+    }
+
+    showErrors(errors);
```

Comment:
Hmmm, this looks like it will be repeated many times in different queries/mutations. Maybe we could extend the base func for making queries to also use alerts? I guess that sometimes we may want to not show alerts so we could pass a bool eg. `showAlerts`. It's just a loose thought 😛

Reply:
Agree, I have implementation partly ready but typing is a trick and will need a refactor so leaving that for another pr